test/test.js

```diff
@@ -0,0 +1,212 @@
+        .then(res => res.data);
+    } catch(err) {
+      should.exist(err);
+      err.response.status.should.equal(500);
+      err.response.data.should.contain('Key not found');
+    }
+    should.not.exist(response);
+  });
+  it('uses `getIdentity` function to retrieve identity', async() => {
+    const requestOptions = {
+      headers: {},
+      method: 'GET',
+      url: `https://localhost:${config.port}/gamma`,
+    };
+    const identity = mockData.identities.alpha;
+    await helpers.createHttpSignatureRequest(
+      {algorithm: 'ed25519', identity, requestOptions});
+    let response;
+    try {
+      response = await axios.get(requestOptions.url, requestOptions)
+        .then(res => res.data);
+    } catch(err) {
+      should.not.exist(err);
+    }
+    response.should.be.an('object');
+    response.success.should.be.true;
+    response.id.should.equal(identity.id);
+    should.exist(response.user);
+    response.user.vip.should.be.true;
+  });
+  it('fails if identity is not found', async() => {
+    const requestOptions = {
+      headers: {},
+      method: 'GET',
+      url: `https://localhost:${config.port}/gamma`,
+    };
+    const identity = mockData.identities.beta;
+    await helpers.createHttpSignatureRequest(
+      {algorithm: 'ed25519', identity, requestOptions});
+    let response;
+    try {
+      response = await axios.get(requestOptions.url, requestOptions)
+        .then(res => res.data);
+    } catch(err) {
+      should.exist(err);
+      err.response.status.should.equal(500);
+      err.response.data.should.contain('User not found');
+    }
+    should.not.exist(response);
+  });
+  it('fails if host header fails validation', async() => {
+    // validator is expecting a `host` of `localhost`
+    const requestOptions = {
+      headers: {},
+      method: 'GET',
+      url: `https://127.0.0.1:${config.port}/alpha`,
+    };
+    const identity = mockData.identities.alpha;
+    await helpers.createHttpSignatureRequest(
+      {algorithm: 'ed25519', identity, requestOptions});
+    let response;
+    try {
+      response = await axios.get(requestOptions.url, requestOptions)
+        .then(res => res.data);
+    } catch(err) {
+      should.exist(err);
+    }
+    should.not.exist(response);
+  });
+  it('fails if publicKey has been revoked', async() => {
+    const requestOptions = {
+      headers: {},
+      method: 'GET',
+      url: `https://localhost:${config.port}/alpha`,
+    };
+    const identity = mockData.identities.delta;
+    await helpers.createHttpSignatureRequest(
+      {algorithm: 'ed25519', identity, requestOptions});
+    let response;
+    try {
+      response = await axios.get(requestOptions.url, requestOptions)
+        .then(res => res.data);
+    } catch(err) {
+      should.exist(err);
+      err.response.status.should.equal(500);
+      err.response.data.should.contain('Public key has been revoked');
+    }
+    should.not.exist(response);
```

Comment:
Related to the revoked test passing, I'm unclear what's happening here.  It's showing up as a race condition for me.  Sometimes passes, sometimes fails here because the response exists.  Console logging seems to "fix" it so unsure where the issue is.

Reply:
Ah, this is a problem due to the test setup.  It creates that revoked property if the mock data rovoked property is true.  But if the test runs fast enough, it will be the same millisecond in the _timestampBeforeNow check, so revoked time is not less than now.

Reply:
Already pushed a fix for this along with regex.